src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs

```diff
@@ -199,14 +241,13 @@ void ExecuteWithAbi (string [] supportedAbis, string apkInputPath, string apkOut
 						count = 0;
 					}
 				}
+				// Clean up Removed files. 
+				foreach (var entry in existingEntries) {
+					Log.LogDebugMessage ($"Removing {entry} as it is not longer required.");
+					apk.Archive.DeleteEntry (entry);
```

Comment:
If we removed files, does this need a `Flush()`?

I vaguely remember having problems here when I didn't `Flush()`: https://github.com/xamarin/xamarin-android/blob/0153c2ceeaaa80b0dacb2f5f840ee1bf5f541f32/src/Xamarin.Android.Build.Tasks/Tasks/BuildBaseAppBundle.cs#L33-L43